common/rpc/encryption/localStorePerHostCertProviderMap.go

```diff
@@ -52,7 +53,10 @@ func newLocalStorePerHostCertProviderMap(overrides map[string]config.ServerTLS,
 
 	for host, settings := range overrides {
 		lcHost := strings.ToLower(host)
-		providerMap.certProviderCache[lcHost] = certProviderFactory(&config.GroupTLS{Server: settings}, nil, nil)
+
+		provider := certProviderFactory(&config.GroupTLS{Server: settings}, nil, nil)
+		providerMap.certProviderCache[lcHost] = provider
+		provider.Initialize(refreshInterval)
```

Comment:
can we just pass in the refreshInterval as a part of the certProviderFactory? Conversely, if a user "forgets" to call Initialize(), does that mean the certProviderFactory "works" but just w/o Refresh? Or will it cause a panic? Can we have a friendly error if someone tries to use the provider w/o initializing it OR can we just avoid the initialize part?

Reply:
even if we do have the initialize() method, the refreshInterval is pretty implementation specific. It makes more sense to have Initialize() take no arguments and have the refreshInterval attached to the object as part of its creation

Reply:
Rearranged and simplified the whole thing.